pkg/icingadb/v1/environment.go

```diff
@@ -1,10 +1,42 @@
 package v1
 
 import (
+	"context"
 	"github.com/icinga/icingadb/pkg/types"
 )
 
 type Environment struct {
 	EntityWithoutChecksum `json:",inline"`
 	Name                  types.String `json:"name"`
 }
+
+// Ctx returns a new Context that carries this Environment as value.
+func (e *Environment) Ctx(parent context.Context) context.Context {
+	return context.WithValue(parent, environmentContextKey, e)
+}
+
+// FromCtx swaps this Environment to the Environment value stored in ctx, if any:
+//
+// 	var e Environment
+// 	if !e.FromCtx(ctx) {
+// 		// Error handling.
+// 	}
+func (e *Environment) FromCtx(ctx context.Context) bool {
```

Comment:
I also looked at the examples, but they have a package scope for the user and we have a broader scope. This is why I added the methods to the env struct to have scope. Otherwise we would have `EnvironmentFromContext` and` NewContextWithEnvironment` or something like that. And if we introduce more of this, we would have more and more unscoped functions. I think it's cleaner this way and it could satisfy an interface too if we ever need to.

Reply:
IMO this is pretty similar to a `func NewEnvironment() *Environment` function, which is also not really scoped and you probably wouldn't make a `func (e *Environment) Init()` instead.

Reply:
I find the comparison a little bit far-fetched as this is a constructor, a single function and in your example is guaranteed to return a new environment while `FromContext` belongs to `NewContext` and extracts an existing environment, if any. And by the loss of scope I mean that one function is then in the package and the other is attached to the struct. For me, both functions belong together and that's why I put both in the same scope. But I can just change that.
Also, I wouldn't just return `* Environment` and require a a non-nil check. To me this is unexpected behavior and the bool clearly indicates that this is an operation that could fail.

Reply:
Ready to review.